src/modules/DialogOverlay.vue

```diff
@@ -105,18 +111,32 @@ export default defineComponent({
     toggleDialog() {
       this.$emit('toggle-dialog')
     },
-    selectProductType(): ProductEnum {
-      const productType = this.tabsData[0].productType
+    getProductType(): ProductEnum {
+      const productType = this.product.productType
       return productType
     },
-    getDataForProduct(): TabItemProps[] {
-      const productType = this.selectProductType()
-      const productData = this.tabsData.find(
-        product => product.productType === productType
-      )
-      if (!productData)
-        throw new Error(`No data found corresponding to ${productType}`)
-      return productData.productItems
+    getProductData(): TabItemProps[] {
+      const productData = this.product.productItems
+      return productData
+    },
+    getOverlayTitle(): string {
+      const productType = this.getProductType()
+      switch (productType) {
+        case ProductEnum.LongTermInterestFree:
+          return 'Long Term Finance Offer'
+        case ProductEnum.PaymentHoliday: {
+          const interestFreePeriod = getProductValueByKey(
+            this.product,
+            'interestFreePeriod'
+          )
+          return `${interestFreePeriod} ${pluralize(
+            interestFreePeriod as number,
+            'Month'
+          )} No Payments & No Interest`
+        }
+        default:
+          return productType
+      }
```

Comment:
new function for getting the overlay title. Design below:
![image](https://user-images.githubusercontent.com/63985172/140860875-fba80433-c4ff-47b9-ace4-a4754b3a73bf.png)
and
![image](https://user-images.githubusercontent.com/63985172/140860908-80226ade-8e79-4ae8-979b-eaea9847e9c4.png)